src/main/java/functions/TreepatFunction.kt

```diff
@@ -0,0 +1,23 @@
+package functions
+
+fun treepatFunction(expression: VisitorFunction): VisitorFunction {
+    return { targetTreeNode ->
+        if (targetTreeNode == null) {
+            VisitorFunctionResponse(listOf(VisitorFunctionSimpleResponse(lastVisitedSibling = targetTreeNode)))
+        }
```

Comment:
It is needed? If the target tree node is null, the other code catch the invalid response

Reply:
Yes, I should get rid of that.